misc/kafka-util/scripts/print_archive.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python3
+#
+# Copyright Materialize, Inc. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# Script to read all messages in an arrow archive of a kafka topic and print the python objects to
+# stdout
+
+import argparse
+import io
+import json
+import logging
+import pathlib
+import pprint
+import sys
+import typing
+
+import avro.io  # type: ignore
+import avro.schema  # type: ignore
+import pyarrow  # type: ignore
+import pyarrow.fs  # type: ignore
+
+# Setup basic formatting for logging output
+logging.basicConfig(format="%(asctime)s %(levelname)s %(name)s %(message)s")
+log = logging.getLogger("print.records")
+log.setLevel(logging.INFO)
+
+
+def decode(reader: avro.io.DatumReader, msg: bytes) -> typing.Any:
+    """Return the Python object represented by the avro-encoded message."""
+    decoder = avro.io.BinaryDecoder(io.BytesIO(msg))
+    return reader.read(decoder)
+
+
+def print_archive(args: argparse.Namespace) -> None:
+    """Print the decoded representation of messages in an archive file."""
+
+    # Why not 'printer = pprint.pprint if args.pretty_print else print'?
+    # Because mypy exits with 'error: Cannot call function of unknown type'
+    def printer(obj: typing.Any) -> None:
+        if args.pretty_print:
+            pprint.pprint(obj)
+        else:
+            print(obj)
+
+    topic = args.topic
+
+    with open("schemas.json") as fd:
+        schemas = json.load(fd)
+
+    try:
+        key_schema = schemas[f"{topic}-key"]["schema"]
+        value_schema = schemas[f"{topic}-value"]["schema"]
+        if args.print_schemas:
+            printer(json.loads(key_schema))
+            printer(json.loads(value_schema))
+    except KeyError:
+        log.error("Failed to locate schema for topic {topic}")
+        raise
```

Comment:
TIOLI: the stringification of a keyerror is just the missing key, so changing this to the following will give you either `{topic}-key` or `{topic}-value` instead of just `{topic}`. Unlikely to matter, but perhaps very interesting:

```suggestion
    except KeyError as e:
        log.error("Failed to locate schema for topic %s", e)
        raise
```

I know it sucks, but it's also idiomatic to use mod-formatting for logs, because that will cause the formatting to not happen unless that log-level is enabled. Also you forgot the `f` part of this f-string.

Reply:
Done!